src/ui/CodeLensProvider.ts

```diff
@@ -77,6 +77,42 @@ export class CodelensProvider implements vscode.CodeLensProvider {
 					this.codeLenses.push(debugTestCodelens);
 				}
 			}
+
+			// The setting for the coverage code lens needs to be switched on for the mechanism
+			// to be enabled
+			if (vscode.workspace.getConfiguration('codelens-kani').get('highlightCoverage', true)) {
+				// Retrieve harness metadata from the tree-sitter which will be used to place the
+				// `Get coverage info` code lens button.
+				const kani_harnesses = await SourceCodeParser.getAttributeFromRustFile(text);
+
+				for (const harness of kani_harnesses) {
+					const harness_name = harness.harnessName;
+
+					if (harness_name === undefined || harness_name === '') {
+						return [];
+					}
```

Comment:
This means we won't highlight anything for any of the remaining harnesses in `kani_harnesses`. Can we instead jump into the next iteration?

Reply:
Great catch! Shall update to skip to next iteration.